docs/migration/v4.md

```diff
@@ -44,4 +44,24 @@ The `AddJob` method of the `IRuntimeRegistry` has been renamed to `Register` to
 ```diff
 - runtimeRegistry.AddJob(r => r.AddJob);
 + runtimeRegistry.Register(r => r.AddJob);
+```
+
```

Comment:
With regards to the RuntimeRegistry, I wonder if we should keep both `Register` and `TryRegister`.

I feel like the impression of choice is biased.
Using `Register` without an inner knowledge of what's at work is running blind with a pair of scissors in hand.

How would you feel about only keeping `TryRegister`?

Independently of your answer, maybe should we add some hint about its existence in the doco. :wink:

Reply:
That is a valid point. Another approach is to allow the user to check if a job already exists.
But with the builder pattern, that feels awkward.

Reply:
We could also do a mixture: We do the following:
```csharp
bool Register(...);
```

Which is a mix of those two. Where we return `false` if a duplicate is detected. Kind of like a `HashSet`

Reply:
See last commit

Reply:
>bool Register(...);

I don't know. Somehow, I wonder if that wouldn't be somehow worse than the `void` alternative. :-/

The returned bool would certainly be ignored (as it's done, most of the time when using the HashSet ;-) )

And my understanding is that this wouldn't throw anymore. So the thing that user was intended to do wouldn't happen and he would have no clue why.

Reply:
Yeah. Just offering "TryRegister" doesn't feel right as well. At least for me.
Maybe we need a "DoesJobExists" - then it is up to the user. But I don't see that is better API wise.

Reply:
I may be a bit (too much?) obsessed with observabilty, but I think that if we grant the user with the possibility to alter the Registry at runtime, we should provide him/her with ways to be able to understand why something didn't work as expected...

In the end, I think I liked the `TryRegister` option better. At least there was an exception that could be potentially processed.

Reply:
Fair. We had some observability planned (well more for the dashboard initially, but hey :D why not reusing it :D )

Before we release `v4` as stable maybe someone has a strong opinion. Let's roll with `TryRegister`

Reply:
The only real bummer here is that you always have to specify the `out` variable. That really bums me more out than it should

Reply:
Yeah even short: `out _` but even that is a lot. But you are right, the overloads might do the trick. Will work on that

Reply:
Said and done! Thanks for the review. I'm loving it